src/rpc/rawtransaction.cpp

```diff
@@ -285,24 +285,35 @@ UniValue verifytxoutproof(const JSONRPCRequest& request)
     if (request.fHelp || request.params.size() != 1)
         throw std::runtime_error(
             "verifytxoutproof \"proof\"\n"
-            "\nVerifies that a proof points to a transaction in a block, returning the transaction it commits to\n"
-            "and throwing an RPC error if the block is not in our best chain\n"
+            "\nVerifies that a proof points to a transaction in a block, returning the block hash and transactions it commits to,\n"
+            "throwing an RPC error if the block is not in our best chain\n"
             "\nArguments:\n"
             "1. \"proof\"    (string, required) The hex-encoded proof generated by gettxoutproof\n"
             "\nResult:\n"
-            "[\"txid\"]      (array, strings) The txid(s) which the proof commits to, or empty array if the proof is invalid\n"
+            "{\n"
+            "    \"blockhash\"   (string) The blockhash the included proof commits to\n"
+            "    [\"txid\"]      (array, strings) The txid(s) which the proof commits to\n"
+            "}\n"
+            "\nResult: (DEPRECATED. To see this result, please start bitcoind with -deprecatedrpc=verifytxoutproof. This option will be removed in v0.18. Empty if proof is invalid)\n"
+            "[\"txid\"]      (array, strings) The txid(s) which the proof commits to\n"
         );
 
     CDataStream ssMB(ParseHexV(request.params[0], "proof"), SER_NETWORK, PROTOCOL_VERSION | SERIALIZE_TRANSACTION_NO_WITNESS);
     CMerkleBlock merkleBlock;
     ssMB >> merkleBlock;
 
-    UniValue res(UniValue::VARR);
+    UniValue res(UniValue::VOBJ);
+    UniValue txids(UniValue::VARR);
 
     std::vector<uint256> vMatch;
     std::vector<unsigned int> vIndex;
-    if (merkleBlock.txn.ExtractMatches(vMatch, vIndex) != merkleBlock.header.hashMerkleRoot)
-        return res;
+    if (merkleBlock.txn.ExtractMatches(vMatch, vIndex) != merkleBlock.header.hashMerkleRoot) {
+        if (IsDeprecatedRPCEnabled("verifytxoutproof")) {
+            return txids;
+        } else {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "No valid transaction commitments are found in the proof");
```

Comment:
Is the RPC user at fault when this happens? If not, I don't think this should be an RPC error.

Reply:
We already error out when the block is not found in the chain, which is just as much the users' fault.

I'll leave this up to others to chime in.

Reply:
Ok, fair. Let's keep it this way then.